cpp/ql/src/Security/CWE/CWE-732/FilePermissions.qll

```diff
@@ -1,6 +1,29 @@
 import cpp
 import semmle.code.cpp.commons.unix.Constants
 
+bindingset[input]
+int parseHex(string input) {
+  input.prefix(2) = "0x" and
+  result =
+    strictsum(int ix |
+      ix in [2 .. input.length()]
+    |
+      16.pow(input.length() - (ix + 1)) * "0123456789abcdef".indexOf(input.charAt(ix).toLowerCase())
+    )
+}
+
+int o_creat_new() {
```

Comment:
This should be named `o_creat`, but that conflicts with `semmle.code.cpp.commons.unix.Constants`. Should this predicate be moved to `Constants` and replace the one there?

Reply:
I'm a bit reluctant about (though not strongly against) changing `semmle.code.cpp.commons.unix.Constants` because its clearly specific to the Unix implementation, where the value given presumably is always correct.

You can solve the name clash by replacing
```
import semmle.code.cpp.commons.unix.Constants
```
with
```
import semmle.code.cpp.commons.unix.Constants as UnixConstants
```
and then accessing it via `UnixConstants::o_creat` (or whatever).

Reply:
Resolved as suggested.